crates/router/src/connector/rapyd/transformers.rs

```diff
@@ -0,0 +1,481 @@
+use error_stack::{IntoReport, ResultExt};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::{
+    core::errors,
+    pii::PeekInterface,
+    services,
+    types::{
+        self, api,
+        storage::enums,
+        transformers::{self, ForeignFrom},
+    },
+};
+
+#[derive(Default, Debug, Serialize)]
+pub struct RapydPaymentsRequest {
+    pub amount: i64,
+    pub currency: enums::Currency,
+    pub payment_method: PaymentMethod,
+    pub payment_method_options: PaymentMethodOptions,
+    pub capture: bool,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct PaymentMethodOptions {
+    #[serde(rename = "3d_required")]
+    pub three_ds: bool,
+}
+#[derive(Default, Debug, Serialize)]
+pub struct PaymentMethod {
+    #[serde(rename = "type")]
+    pub pm_type: String,
+    pub fields: PaymentFields,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct PaymentFields {
+    pub number: String,
+    pub expiration_month: String,
+    pub expiration_year: String,
+    pub name: String,
+    pub cvv: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for RapydPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        match item.request.payment_method_data {
+            api_models::payments::PaymentMethod::Card(ref ccard) => {
+                let payment_method = PaymentMethod {
+                    pm_type: "in_amex_card".to_owned(), //https://github.com/juspay/hyperswitch/issues/369
+                    fields: PaymentFields {
+                        number: ccard.card_number.peek().to_string(),
```

Comment:
Keep the connector field also secret itself. our serializer will take care of it.

Reply:
cool